x-pack/plugins/security_solution/server/lib/telemetry/artifact.ts

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import axios from 'axios';
+import AdmZip from 'adm-zip';
+import type { ITelemetryReceiver } from './receiver';
+import type { ESClusterInfo } from './types';
+
+export interface IArtifact {
+  start(receiver: ITelemetryReceiver): Promise<void>;
+  getArtifact(name: string): Promise<unknown>;
```

Comment:
Picking nits now, but why `Promise<unknown>`
We know the data shape is:

```ts
{
    "artifacts": {
        [artifact_name]: {
            decoded_sha256: string,
            decoded_size: number,
            relative_url: string
        }
    },
    "manifest_version": "1.0.3",
    "schema_version": "1.0.0"
}
```

Might as well harden this. I see you are:

```ts
as unknown as TelemetryConfiguration
```